api4/post.go

```diff
@@ -485,6 +486,23 @@ func getPostsByIds(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getEditHistoryForPost(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequirePostId()
+	if c.Err != nil {
+		return
+	}
+
+	postsList, err := c.App.GetEditHistoryForPost(c.Params.PostId)
```

Comment:
I think we should have some permission check here? I don't know too much about the feature but is it allowed to get post history for everyone?

Reply:
Good point related to permission checks. 

I believe permission checks should be same as editing post. If a user has permission to edit a post, they should be also able to restore an older version. I am leaving below permission related piece from `patchPost`. I wait for the final decision  from @esethna

https://github.com/mattermost/mattermost-server/blob/0c3704c9ca6ab30f7e643de7ff2ee45e34ae7925/api4/post.go#L816-L826

Reply:
@sinansonmez @isacikgoz thanks for the question re permission checks.

- I agree that any user that has edit permissions on the post should be able to restore previous versions. 
- However, I think any user (even if they don't have edit permissions on the post) should be able to VIEW the edit history.

Reply:
For @isacikgoz information, I copying Eric's latest guidance here https://github.com/mattermost/mattermost-webapp/pull/11069#issuecomment-1256838124

Reply:
Thanks for the clarification and the info 💯